### PR TITLE
Fixed TO log warning when generating snapshots for topology-based delivery services

### LINKED ISSUE #5192: TO logs warnings when generating snapshots for topology-based delivery services
## I'm submitting a ...
-  bug report

## Traffic Control components affected ...
-  Traffic Ops

## Current behavior:
Traffic Ops logs a warning when generating snapshots for topology-based delivery services:
```
WARNING: servers.go:364: 2020-10-26T13:35:43.892585-06:00: Creating CRConfig: deliveryservice rawl-http has no regexes, skipping
``` 

## Expected behavior:
Traffic Ops should not log this warning when generating snapshots for topology-based delivery services.

## Minimal reproduction of the problem with instructions:
Create a topology. Assign it to a delivery service. Snapshot the delivery service's CDN.

### PR DESCRIPTION
**Pull request by rimashah25**
## What does this PR (Pull Request) do?

- [x] This PR fixes #5192 

## Which Traffic Control components are affected by this PR?

- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?

Run TP end-to-end tests and TO API tests

## If this is a bug fix, what versions of Traffic Control are affected?

- master(d9d67cf)

## The following criteria are ALL met by this PR

- [x] I have explained why tests are unnecessary
- [x] I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information

The PR doesn't have tests or documentation update as the logic hasn't changed.

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/crconfig/servers.go [hunk 1]
```diff
@@ -361,7 +361,6 @@ order by dsr.set_number asc
 		for _, dsName := range dses {
 			dsInfList, ok := dsInfs[string(dsName)]
 			if !ok {
-				log.Warnln("Creating CRConfig: deliveryservice " + string(dsName) + " has no regexes, skipping")
```
Comment (zrhoffman) [inline traffic_ops/traffic_ops_golang/crconfig/servers.go:364]: We only want to remove this warning for topology-based delivery services. Other DSes should still have the warning.

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: traffic_ops/traffic_ops_golang/crconfig/servers.go (hunk 1 above)
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/crconfig/servers.go:364]: ah, this answers my question on the related Issue

Comment (rimashah25) [reply traffic_ops/traffic_ops_golang/crconfig/servers.go:364]: Done.

**Review by ocket8888 — APPROVED** [review]
Confirmed this fixes the issue, and Zach said it was good to merge once I say so, and the tests pass so I'm saying so.

**Pull request merged**